pkg/ottl/contexts/internal/ctxutil/slice.go

```diff
@@ -5,61 +5,258 @@ package ctxutil // import "github.com/open-telemetry/opentelemetry-collector-con
+			}
+			raw[i] = sv
 		}
-		i = resInt
+		s.FromRaw(raw)
+	case pcommon.Slice:
+		raw := make([]V, typeVal.Len())
+		for i := 0; i < typeVal.Len(); i++ {
+			v, ok := typeVal.At(i).AsRaw().(V)
+			if !ok {
+				return fmt.Errorf("invalid value type provided for a slice of %T: %T", raw, typeVal.At(i).AsRaw())
+			}
+			raw[i] = v
+		}
+		s.FromRaw(raw)
+	default:
+		return fmt.Errorf("invalid type provided for setting a slice of %T: %T", val, *new(V))
 	}
 
-	idx := int(*i)
+	return nil
+}
 
-	if idx < 0 || idx >= s.Len() {
-		return fmt.Errorf("index %d out of bounds", idx)
+// GetCommonIntSliceValues converts a pdata typed slice of [constraints.Integer] into
+// []int64, which is the standard OTTL type for integer slices.
+func GetCommonIntSliceValues[V constraints.Integer](val CommonTypedSlice[V]) []int64 {
+	output := make([]int64, val.Len())
+	for i := 0; i < val.Len(); i++ {
+		output[i] = int64(val.At(i))
 	}
+	return output
+}
 
-	return SetIndexableValue[K](ctx, tCtx, s.At(idx), val, keys[1:])
+// GetCommonIntSliceValue is like GetCommonTypedSliceValue, but for integer pdata typed
+// slices.
+func GetCommonIntSliceValue[K any, V constraints.Integer](ctx context.Context, tCtx K, s CommonTypedSlice[V], keys []ottl.Key[K]) (int64, error) {
+	value, err := GetCommonTypedSliceValue[K, V](ctx, tCtx, s, keys)
+	if err != nil {
+		return 0, err
+	}
+	return int64(value), nil
+}
+
+// SetCommonIntSliceValue is like SetCommonTypedSliceValue, but for integer pdata typed
+// slice element values. [V] is the type of the slice elements.
+// The any-wrapped value type must be and integer convertible to [V], otherwise an error
+// is returned. This function does not support slice elements indexing.
+func SetCommonIntSliceValue[K any, V constraints.Integer](ctx context.Context, tCtx K, s CommonTypedSlice[V], keys []ottl.Key[K], val any) error {
+	var intVal V
+	switch typeVal := val.(type) {
+	case int:
+		intVal = V(typeVal)
+	case int8:
+		intVal = V(typeVal)
+	case int16:
+		intVal = V(typeVal)
+	case int32:
+		intVal = V(typeVal)
+	case int64:
+		intVal = V(typeVal)
+	case uint:
+		intVal = V(typeVal)
+	case uint8:
+		intVal = V(typeVal)
+	case uint16:
+		intVal = V(typeVal)
+	case uint32:
+		intVal = V(typeVal)
+	case uint64:
+		intVal = V(typeVal)
+	default:
+		return fmt.Errorf("invalid type provided for setting a slice of %T: %T", *new(V), val)
+	}
+
+	return SetCommonTypedSliceValue[K, V](ctx, tCtx, s, keys, intVal)
+}
+
+// SetCommonIntSliceValues is like SetCommonTypedSliceValues, but for integer pdata typed
+// slices. [V] is the type of the slice elements. The value must be []any, []int64, []T,
+// or a pcommon.Slice which elements are type inferable to int64, otherwise an error is
+// returned.
+func SetCommonIntSliceValues[V constraints.Integer](s CommonTypedSlice[V], val any) error {
+	switch typeVal := val.(type) {
+	case CommonTypedSlice[V]:
+		s.FromRaw(typeVal.AsRaw())
+	case []int64:
```

Comment:
Do we want cases for []int32 and []int as common cases?

Reply:
I think we shouldn't. OTTL should only generate slices of []int64, and other integer types here would probably mean we've an error. Even if we handle them here, it might be causing other issues as several functionalities assumes integers are int64.